src/write.c

```diff
@@ -567,15 +567,23 @@ static avifResult avifEncoderAddImageInternal(avifEncoder * encoder,
         return AVIF_RESULT_NO_CONTENT;
     }
 
-    if ((cellCount > 1) && ((firstCell->width < 64) || (firstCell->height < 64))) {
+    if (!avifAreGridDimensionsValid(firstCell->yuvFormat, gridCols, gridRows, firstCell->width, firstCell->height, &encoder->diag)) {
         return AVIF_RESULT_INVALID_IMAGE_GRID;
     }
 
+    avifBool firstTileUVPresent = (firstCell->yuvPlanes[AVIF_CHAN_U] && firstCell->yuvPlanes[AVIF_CHAN_V]);
```

Comment:
It is not necessary to check `uvPresent != firstTileUVPresent`, because this check is implied by the `cellImage->yuvFormat != firstCell->yuvFormat` check.

Note that `uvPresent` is true if and only if `cellImage->yuvFormat != AVIF_PIXEL_FORMAT_YUV400`.